### PR TITLE
WIFI/ENDPOINT SETUP

### PR DESCRIPTION
**Pull request by siorekoskar**
Initial wifi setup on softAP and first endpoint

### CODE REVIEW

File: platformio.ini [hunk 1]
```diff
@@ -12,3 +12,15 @@
...
+test_ignore = test_endpoint_dispatcher
+monitor_speed = 9600
+
+[env:native]
+platform = native
+
+lib_ldf_mode = deep+
+
+lib_deps = 
+    ESP Async WebServer
```
Comment (Mefffiu) [inline platformio.ini]: Can be removed.

File: src/main.cpp [hunk 2]
```diff
@@ -1,9 +1,44 @@
...
-  // put your setup code here, to run once:
+#include <ESP8266WiFi.h>
+#include <ESP8266WebServer.h>
+#include <WiFiClient.h>
+#include <ESP8266mDNS.h>
+
+ESP8266WebServer server(80);
+
+void handleRoot()
+{
```
Comment (Mefffiu) [inline src/main.cpp]: `{` should be in the same line as `void handleRoot()`

File: src/main.cpp [hunk 3]
```diff
@@ -1,9 +1,44 @@
...
+  server.send(200, "text/plain", "hello from esp8266!");
+}
+
+void setup()
```
Comment (Mefffiu) [inline src/main.cpp]: `{` should be in the same line as `void setup()`

File: src/main.cpp [hunk 4]
```diff
@@ -1,9 +1,44 @@
...
+  IPAddress subnet(255, 255, 255, 0);
+  WiFi.config(ip, gateway, subnet);
+  Serial.begin(9600);
+  Serial.println();
+  Serial.println(WiFi.localIP());
+
+  Serial.print("Setting soft-AP ... ");
+  boolean result = WiFi.softAP("ESPSKAR", "password");
+  if (result == true)
+  {
```
Comment (Mefffiu) [inline src/main.cpp]: `{` should be in line 26

File: src/main.cpp [hunk 5]
```diff
@@ -1,9 +1,44 @@
...
+    Serial.println("Ready");
+  }
+  else
+  {
```
Comment (Mefffiu) [inline src/main.cpp]: `{` should be in line 31

File: src/main.cpp [hunk 6]
```diff
@@ -1,9 +1,44 @@
...
+
+  server.on("/", handleRoot);
+  server.begin();
+  Serial.println("HTTP server started");
 }
 
-void loop() {
-  // put your main code here, to run repeatedly:
+void loop()
+{
```
Comment (Mefffiu) [inline src/main.cpp]: `{` should be in line 40

**Review by Mefffiu — CHANGES_REQUESTED** [review]
Please, remove unused dependencies and apply shared coding rules.

File: src/main.cpp (hunk 6 above)
Comment (siorekoskar) [reply src/main.cpp]: done

File: src/main.cpp (hunk 5 above)
Comment (siorekoskar) [reply src/main.cpp]: done

File: src/main.cpp (hunk 4 above)
Comment (siorekoskar) [reply src/main.cpp]: done

File: src/main.cpp (hunk 3 above)
Comment (siorekoskar) [reply src/main.cpp]: done

File: src/main.cpp (hunk 2 above)
Comment (siorekoskar) [reply src/main.cpp]: done

File: platformio.ini (hunk 1 above)
Comment (siorekoskar) [reply platformio.ini]: done

**Pull request merged**